experimental/qnamaker-prompting/javascript_nodejs/dialogs/functionDialogBase.js

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { Dialog, DialogTurnStatus, DialogSet } = require('botbuilder-dialogs');
+const { ActivityTypes } = require('botbuilder');
+
+const functionStateName = 'functionState';
+
+class FunctionDialogBase extends Dialog {
+    
+    constructor(dialogId) {
+        super(dialogId);
+    }
+
+    async beginDialog(dc) {
+        // Don't do anything for non-message activities.
+        if (dc.context.activity.type !== ActivityTypes.Message) {
+            return Dialog.EndOfTurn;
+        }
+
+        // Run dialog logic.
+        return await this.runStateMachine(dc);
+    }
+
+    async continueDialog(dc) {
+        // Skip non-message activities.
+        if (dc.context.activity.type !== ActivityTypes.Message) {
+            return Dialog.EndOfTurn;
+        }
+
+        // Run dialog logic.
+        return await this.runStateMachine(dc);
+    }
+
+    getPersistedState(dialogInstance) {
+        return dialogInstance.state[functionStateName];
+    }
+
+    async runStateMachine(dc) {
+       
+        var oldState = this.getPersistedState(dc.activeDialog);
+                
+        var input = dc.context.activity.text;
```

Comment:
this line isn't necessary...

and just took a quick look at the C# and the same issue is there - can you remove it from both.

Reply:
Done.